api.json

```diff
@@ -0,0 +1,802 @@
+                                },
+                                "examples": {
+                                    "invalid deku address": {
+                                        "$ref": "#/components/examples/invalid-deku-address-error"
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        },
+        "/states/wasm": {
+            "get": {
+                "description": "Returns the whole state of the wasm vm.",
+                "tags": [
+                    "wasm"
+                ],
+                "responses": {
+                    "200": {
+                        "description": "Wasm state retrieved",
+                        "content": {
+                            "application/json": {
+                                "schema": {
+                                    "type": "object"
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        },
+        "/states/unix": {
+            "get": {
+                "description": "Returns the chole state of the deku-p state",
+                "tags": [
+                    "deku-p"
+                ],
+                "responses": {
+                    "200": {
+                        "description": "Deku-p state retrieved",
+                        "content": {
+                            "application/json": {
+                                "schema": {
+                                    "type": "object"
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        },
+        "/states/unix/{key}": {
+            "get": {
+                "description": "Returns a subpart of the state of deku-p, default is null",
+                "tags": [
+                    "deku-p"
+                ],
+                "parameters": [
+                    {
+                        "name": "key",
+                        "in": "path",
+                        "schema": {
+                            "type": "string"
+                        },
+                        "required": true,
+                        "description": "the key of the json state",
+                        "example": "custom_key"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "Deku-p state retrieved.",
+                        "content": {
+                            "application/json": {
+                                "schema": {
+                                    "type": "object"
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        },
+        "/helpers/hash-operation" : {
```

Comment:
I think this is great to provide them to help our clients. But I think it is way more great, to hash the operation for the client under the hood.
I feel like the only use cases for this one, will be:
- oh, I forgot what was the hash of this operation (almost impossible, because I would need to remember the details of each operation I submitted)
- I don't know how to hash, please tell me what would be the hash of this operation I want to submit.
If the second use case looks more realistic, it could lead to have some clients performing 3 HTTP request to submit an operation:
- retrieve block level
- get correct hash
- submit operation
We should be careful on that, because it could add a lot of load on the nodes.

Reply:
Yes, it's mainly for the second use case: a client doesn't know how to hash the operation/packet (because there isn't any hashing library in its favorite language for instance).
It's just a helper, you don't have to use it, for instance in the deku-toolkit I am writing in typescript does not relay on this endpoint, it hashes the payload in the client side